### PR TITLE
Remove duplicate logic with invalid validation

### PR DESCRIPTION
**Pull request by philipsd6**
Existing logic hid the correct behavior of requiring either `template` or `template_url` _only_ if state='present'.

### CODE REVIEW

**Comment by robynbergeron** [comment]
Thanks @philipsd6. @jsmartin please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by jsmartin** [comment]
needs_revision.  @philipsd6 can you squash this into a single commit?

**Comment by philipsd6** [comment]
@jsmartin Squashed as requested. 

ready_for_review

**Comment by philipsd6** [comment]
@jsmartin Can you PTAL, or give me an estimate? This should be ready to go…

File: cloud/amazon/cloudformation.py [hunk 1]
```diff
@@ -258,15 +258,12 @@ def main():
...
-    if module.params['template'] is None and module.params['template_url'] is None:
-        module.fail_json(msg='Either template or template_url expected')
-
     state = module.params['state']
     stack_name = module.params['stack_name']
 
     if module.params['template'] is None and module.params['template_url'] is None:
         if state == 'present':
-            module.fail_json('Module parameter "template" or "template_url" is required if "state" is "present"')
+            module.fail_json(msg='Module parameter "template" or "template_url" is required if "state" is "present"')
```
Comment (bcoca) [inline cloud/amazon/cloudformation.py:266]: ~~actually, just use `module.required_one_of` function~~ nvmd, this is only required when present now.

**Comment by gregdek** [comment]
@philipsd6 After some review, it looks as though this PR needs some revisions. 

Please make the suggested revisions, and when you're done, please add a comment with the text "ready_for_review" and we will continue with the review process.  Thanks!

**Comment by gregdek** [comment]
@philipsd6 A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@philipsd6 Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@philipsd6 A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by philipsd6** [comment]
@bcoca rescinded his comment, so AFAICT no revision is necessary.

ready_for_review

File: cloud/amazon/cloudformation.py (hunk 1 above)
Comment (bcoca) [inline cloud/amazon/cloudformation.py:266]: but there is`equired_if`, something like:

```
module.required_if(['state', 'present', ['template']])
```

**Comment by gregdek** [comment]
Thanks @philipsd6 for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@philipsd6 A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@philipsd6 Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**